CISM support for NorESM diagnostics

hgoelzer opened the pull request:
This pull request adds basic CISM support to the NorESM diagnostics package. CISM diagnostic output can be selected as an additional component cism similar to the other components.

hgoelzer committed:
Adding rudimentary CISM diagnostic support

hgoelzer committed:
fixed timers to work with cism end of year notation
included original scalars
switched all to indexnew and renamed to index

hgoelzer committed:
Merge branch 'NordicESMhub:master' into master

hgoelzer committed:
merging back netcdf fix

hgoelzer committed:
Merge branch 'NordicESMhub:master' into master

hgoelzer committed:
Update README.md

hgoelzer committed:
Update README.md

hgoelzer committed:
Update diag_run

hgoelzer committed:
Update cism_diag_template.sh

YanchunHe commented on bin/diag_run:
```diff
@@ -384,6 +389,8 @@ else
             let RUN_MODELS[3]=${RUN_MODELS[3]}+1
         elif [ "$model" == "hamocc" ]; then
             let RUN_MODELS[4]=${RUN_MODELS[4]}+1
+        elif [ "$model" == "cism" ] || [ "$model" == "cism" ]; then
```
change

```bash
elif [ "$model" == "cism" ] || [ "$model" == "cism" ]; then
```

to
```bash
elif [ "$model" == "cism" ]; then
```

??

YanchunHe commented on bin/diag_run:
```diff
@@ -221,7 +223,7 @@ else
                 else
                     echo "ERROR: no model specified (-m, --model)"
-                         echo "Valid options: cam, clm, cice, all"
+                         echo "Valid options: cam, clm, cice, cism, all"
```
"Valid options: cam, blom/micom,clm, cice, cism, all"

YanchunHe commented on packages/CAM_DIAG/amwg_template.csh:
```diff
@@ -1,4 +1,4 @@
-#!/bin/csh
+#!/bin/csh -e
```
Usually, we want to keep the scrip running even if there are errors.
So I would suggest to remove '-e' option.

YanchunHe commented on packages/CAM_DIAG/amwg_template.csh:
```diff
@@ -11,6 +11,7 @@ setenv DIAG_VERSION 140804  # version number YYMMDD
 if ( -d /opt/ncl65 && -d /opt/nco475 && -d /opt/cdo197 ) then
     setenv NCARG_ROOT /opt/ncl65
     setenv NCARG_COLORMAPS $NCARG_ROOT/lib/ncarg/colormaps
+#    setenv PATH /usr/bin:/usr/local/bin:/usr/bin:/opt/ncl65/bin/:/opt/nco475/bin/:/opt/cdo197/bin
```
I see this is for debug only, so the comment can be removed.

YanchunHe commented on packages/CAM_DIAG/amwg_template.csh:
```diff
@@ -33,6 +34,10 @@ if ( $rcode != 0 ) then
 
+# hg workaround for cdo errors
+echo ncks: $ncksbin nco: $nco_dir cdo: $cdo_dir
+    
```
clean up?

I don't there is problem with 'nco' path?

YanchunHe reviewed (changes requested):
Some small revision of the code need to be done.

hgoelzer committed:
Merge branch 'NordicESMhub:master' into master

hgoelzer committed:
Reverting changes in amwg_template.csh and minor corrections in diag_run following comments from Yanchun

hgoelzer committed:
Merge branch 'master' of github.com:hgoelzer/noresmdiagnostics

hgoelzer commented:
Hi Yanchun,
I have reverted the changes in amwg_template.csh and corrected the issues you marked in diag_run. I hope the pull request can go forward now. Thanks
    Heiko

YanchunHe commented:
Look good, I merge the changes. Thanks a lot, Heiko!
